test/src/SpendPermissions/replace.t.sol

```diff
@@ -0,0 +1,281 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import {SpendPermissionManager} from "../../../src/SpendPermissionManager.sol";
+
+import {SpendPermissionManagerBase} from "../../base/SpendPermissionManagerBase.sol";
+
+contract ReplaceTest is SpendPermissionManagerBase {
+    SpendPermissionManager.SpendPermission existingSpendPermission;
+    SpendPermissionManager.PeriodSpend lastValidUpdatedPeriod;
+
+    function setUp() public {
+        _initializeSpendPermissionManager();
+        vm.prank(owner);
+        account.addOwnerAddress(address(mockSpendPermissionManager));
+
+        // establish an existing spend permission and capture the last valid updated period
+        existingSpendPermission = _createSpendPermission();
+        vm.prank(address(account));
+        mockSpendPermissionManager.approve(existingSpendPermission);
+        lastValidUpdatedPeriod = mockSpendPermissionManager.getLastUpdatedPeriod(existingSpendPermission);
+    }
+
+    function test_replace_revert_invalidSender(
+        address sender,
+        address spender,
+        uint48 start,
+        uint48 end,
+        uint48 period,
+        uint160 allowance,
+        uint256 salt,
+        bytes memory extraData
+    ) public {
+        vm.assume(sender != address(0));
+        vm.assume(sender != address(account));
+        SpendPermissionManager.SpendPermission memory newSpendPermission = SpendPermissionManager.SpendPermission({
+            account: address(account),
+            spender: spender,
+            token: NATIVE_TOKEN,
+            start: start,
+            end: end,
+            period: period,
+            allowance: allowance,
+            salt: salt,
+            extraData: extraData
+        });
+        vm.startPrank(sender);
+        vm.expectRevert(abi.encodeWithSelector(SpendPermissionManager.InvalidSender.selector, sender, account));
+        mockSpendPermissionManager.replace(existingSpendPermission, newSpendPermission, lastValidUpdatedPeriod);
+        vm.stopPrank();
+    }
+
+    function test_replace_revert_invalidLastUpdatedPeriod(
```

Comment:
Can we check more variants of this explicitly? For example, it should fail both if we're in the same period with incremented spend or same spend in a different period boundary?